### PR TITLE
CUMULUS-3806-3 -- Remove internal rec report references/cleanup

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-3806](https://bugs.earthdata.nasa.gov/browse/CUMULUS-3806)

PR No. 3 for this ticket, blocked on the first two PRs.

## Changes

*  Removes deprecated/unused code

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: example/spec/parallel/createReconciliationReport/CreateReconciliationReportSpec.js [hunk 1]
```diff
@@ -734,7 +734,6 @@ describe('When there are granule differences and granule reconciliation is run',
     });
   });
 
-  // TODO: fix tests in CUMULUS-3806 when CreateReconciliationReport lambda is changed to query postgres
```
Comment (etcart) [inline example/spec/parallel/createReconciliationReport/CreateReconciliationReportSpec.js:737]: were these tests fixed incidentally by the shift to Postgres?

File: packages/api/lambdas/create-reconciliation-report.js [hunk 2]
```diff
@@ -967,7 +967,6 @@ async function processRequest(params) {
       log.error(
         'Internal Reconciliation Reports are no longer valid, as Cumulus is no longer utilizing Elasticsearch'
       );
-      //TODO remove internal rec report code
```
Comment (etcart) [inline packages/api/lambdas/create-reconciliation-report.js:970]: was the actual internal rec report code removed in a previous PR and this TODO is vestigial? I checked that it is actually not referenced in the code but I just want to double check what happened here

**Review by etcart — APPROVED** [review]
there are two clarification questions here but my check of the code suggests that answer to both is "yes"

File: packages/api/lambdas/create-reconciliation-report.js (hunk 2 above)
Comment (Jkovarik) [reply packages/api/lambdas/create-reconciliation-report.js:970]: This note is a good portion of what this PR is doing - we removed the call *to* the rec report code in PR 1, but some of the methods referenced were still used until PR2.      This PR is to clean up/remove the unused internal rec report and associated lib methods.

File: example/spec/parallel/createReconciliationReport/CreateReconciliationReportSpec.js (hunk 1 above)
Comment (Jkovarik) [reply example/spec/parallel/createReconciliationReport/CreateReconciliationReportSpec.js:737]: Fixed yes, but intentionally.    This comment just wasn't cleaned up in the prior PRs.

**Pull request merged**